python/ray/experimental/worker.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ray
+import threading
+import numpy as np
+import pyarrow.plasma as plasma
+
+PYTHON_MODE = 2
+
+
+def check_main_thread():
+    """Check that we are currently on the main thread.
+
+    Raises:
+        Exception: An exception is raised if this is called on a thread other
+            than the main thread.
+    """
+    if threading.current_thread().getName() != "MainThread":
+        raise Exception("The Ray methods are not thread safe and must be "
+                        "called from the main thread. This method was called "
+                        "from thread {}."
+                        .format(threading.current_thread().getName()))
+
+
+def get(object_ids, worker=None):
```

Comment:
@robertnishihara: The function names here can be changed, if needed. I was unsure what to call them so have kept it as `get` and `wait` for now.